### ISSUE TITLE
Deprecation warnings since Typst 0.12 (use of measure and style)

### ISSUE THREAD

**Issue by brynhum** [Jollywatt/typst-fletcher#55]
Fletcher 0.5.1 now generates warnings with Typst 0.12, sometimes directly from code in fletcher, and sometimes from code in cetz:0.2.2. Presumably the latter error can be fixed by making it depend on cetz:0.3.1 instead, and the former warnings are related to code in fletcher itself.

```
warning: `style` is deprecated
    ┌─ @preview/fletcher:0.5.1/src/diagram.typ:482:5
    │
482 │     box(style(styles => {
    │ ╭───────^
483 │ │     let options = options
484 │ │
485 │ │     options.em-size = measure(h(1em)).width
    · │
543 │ │     render(grid, nodes, edges, options)
544 │ │   }))
    │ ╰────^
    │
    = hint: use a `context` expression instead
```

```
warning: calling `measure` with a styles argument is deprecated
    ┌─ @preview/fletcher:0.5.1/src/node.typ:357:24
    │
357 │       let (width, height) = measure(box(
    │ ╭───────────────────────────^
358 │ │       node.label,
359 │ │       width:  inner-size.at(0),
360 │ │       height: inner-size.at(1),
361 │ │     ), styles)
    │ ╰──────────────^
    │
    = hint: try removing the styles argument
```

**Comment by NiceWaffel** [comment]
The deprecation warnings in fletcher seem to be fixed with Pull Request #51, not the ones from the outdated CeTZ though.

@Jollywatt when do you plan to release the next package version, so these fixes are generally available?

**Comment by Jollywatt** [comment]
@NiceWaffel I've looked at upgrading to cetz 0.3.0 but it introduces some breaking changes which require some time. I'm a starting PhD student at present so have very little time unfortunately!

**Comment by aaron-jack-manning** [comment]
@Jollywatt Also note that Cetz changed licenses in the update latest update, https://github.com/cetz-package/cetz/pull/598. I will have a bit of available time soon and am happy to contribute to this, but I would need clarification on if fletcher is happy to go GPL too then.

**Comment by Jollywatt** [comment]
I've finally had some time to fix. The breakages I observed with cetz 0.3.0 were luckily fixed in cetz 0.3.1, so it wasn't too bad to update. Fletcher 0.5.2 is now available and requires typst >=0.12.0.
